tests/python_functional/functional_tests/parsers/app-parser/test_topic_syslog.py

```diff
@@ -0,0 +1,44 @@
+    (r"""<14>1 2019-04-12T14:29:06.576Z - RT_FLOW - RT_FLOW_SESSION_CLOSE [junos@2636.1.1.1.2.129 reason=\"TCP FIN\" source-address=\"10.3.255.203\" source-port=\"47776\" destination-address=\"8.23.224.110\" destination-port=\"80\" connection-tag=\"0\" service-name=\"junos-http\" nat-source-address=\"10.3.136.49\" nat-source-port=\"19162\" nat-destination-address=\"8.23.224.110\" nat-destination-port=\"80\" nat-connection-tag=\"0\" src-nat-rule-type=\"source rule\" src-nat-rule-name=\"nat1\" dst-nat-rule-type=\"N/A\" dst-nat-rule-name=\"N/A\" protocol-id=\"6\" policy-name=\"permit_all\" source-zone-name=\"trust\" destination-zone-name=\"untrust\" session-id-32=\"5\" packets-from-client=\"6\" bytes-from-client=\"337\" packets-from-server=\"4\" bytes-from-server=\"535\" elapsed-time=\"1\" application=\"HTTP\" nested-application=\"UNKNOWN\" username=\"N/A\" roles=\"N/A\" packet-incoming-interface=\"ge-0/0/0.0\" encrypted=\"No\" application-category=\"Web\" application-sub-category=\"N/A\" application-risk=\"4\" application-characteristics=\"Can Leak Information;Supports File Transfer;Prone to Misuse;Known Vulnerabilities;Carrier of Malware;Capable of Tunneling;\"] session closed TCP FIN: 10.3.255.203/47776->8.23.224.110/80 0x0 junos-http 10.3.136.49/19162->8.23.224.110/80 0x0 source rule nat1 N/A N/A 6 permit_all trust untrust 5 6(337) 4(535) 1 HTTP UNKNOWN N/A(N/A) ge-0/0/0.0 No Web N/A 4 Can Leak Information;Supports File Transfer;Prone to Misuse;Known Vulnerabilities;Carrier of Malware;Capable of Tunneling;""", "${.app.name}\\n", "junos\n"),
+    ("Apr 18 12:33:50 nobody sudo[25173]:   nobody : TTY=pts/13 ; PWD=/home/nobody ; USER=root ; COMMAND=/bin/ls", "${.app.name}\\n", "sudo\n"),
+    (r"""<0>1 2012-03-05T15:10:34+02:00 localhost kernel 1234 - - @cim: {\"name1\":\"value1\", \"name2\":\"value2\"}""", "${.app.name}\\n", "cim\n"),
+]
+@pytest.mark.parametrize("input_message, template, expected_value", test_parameters_syslog,
+                         ids=list(map(str, range(len(test_parameters_syslog)))))
+def test_application_syslog(config, syslog_ng, input_message, template, expected_value):
+    config.add_include("scl.conf")
+
+    generator_source = config.create_example_msg_generator(num=1, template="\"" + input_message + "\"")
+    syslog_parser = config.create_syslog_parser(flags="syslog-protocol")
```

Comment:
Combined with testing each parser, having this classification and do an end-to-end with default-network driver I think it should cover the majority, meaning this test is imho fine only testing classification; and the result of as such the `template` parameter could have been part of the test itself as fixed value. But this is more like a note, not really a change request, not even an optional one.